### ISSUE TITLE
[Bug][Azure Devops] Grafana panel empty (step3 - check pull request)

### ISSUE THREAD

**Issue by klemen-df** [apache/devlake#5897]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I opened 'DORA Validation dashboard' and there's an issue at retrieving graph 'Step 3. Check if a `pull request` is associated with the correct `depoloyment commit` in table project_pr_metrics'.

I tried to manually execute the first part of mysql:

```
	SELECT
		ppm.id AS pr_id,
		ppm.deployment_commit_id AS id_1,
		pr.merge_commit_sha,
		ppm.project_name 
	FROM
		project_pr_metrics ppm
		LEFT JOIN pull_requests pr ON ppm.id = pr.id 
```

This is what I get:
```
+-------------------------------------+--------------------------------------------------------------------------------+------------------------------------------+--------------+
| pr_id                               | id_1                                                                           | merge_commit_sha                         | project_name |
+-------------------------------------+--------------------------------------------------------------------------------+------------------------------------------+--------------+
| azuredevops:GitPullRequest:1:191898 |                                                                                | 004f9da27f875cade1767f91ce640886f5941f35 | MyProject|
```

You can clearly see that the id_1 is empty.

In Grafana 'pr_id' is queried with:
```
select concat(Url, '--', id) from pull_requests
```

and the URL is retrieved as:
```
select concat(Url, '--', id) from pull_requests

https://dev.azure.com/danfoss/<id>/_apis/git/repositories/<id>/pullRequests/191898
```

This also means that 'ppm.id' and 'pd_id' will never match:
```
	WHERE
		ppm.project_name in ($project) 
		AND ppm.id = '$pr_id'
```

What am I doing wrong?

### What do you expect to happen

Panel in Grafana dashboard should be populated

### How to reproduce

Install devlake 0.17, upgrade to 0.18, connect to Azure Devops, ... nothing non-standard.

### Version

v0.18-beta5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
@KlemenDanfoss Do you have any records in the `cicd_deployment_commits` table?
@CamilleTeruel Are we converting data from the azdo to the table?

**Comment by KlemenDanfoss** [comment]
@klesh , yes

1671 rows, 1300 rows where result='SUCCESS'

**Comment by KlemenDanfoss** [comment]
@klesh 

`mysql> select * from cicd_deployment_commits limit 10;
+--------------------------------------------------------------------------------+-------------------------+-------------------------+--------------------------------------------------------------------------+--------------
-----------+--------------+-------------------------------------+------------------------------------------------------------------+-----------------------------+------------------+---------+--------+-------------+---------
----------------+-------------------------+-------------------------+--------------+------------------------------------------+---------------------------------------------------+--------------------------------------------
----------------------+----------------------------------------------------+--------------------------------------------------------------------------------+
| id                                                                             | created_at              | updated_at              | _raw_data_params                                                         | _raw_data_tab
le         | _raw_data_id | _raw_data_remark                    | cicd_scope_id                                                    | cicd_deployment_id          | name             | result  | status | environment | created_
date            | started_date            | finished_date           | duration_sec | commit_sha                               | ref_name                                          | repo_id
                      | repo_url                                           | prev_success_deployment_commit_id                                              |
+--------------------------------------------------------------------------------+-------------------------+-------------------------+--------------------------------------------------------------------------+--------------
-----------+--------------+-------------------------------------+------------------------------------------------------------------+-----------------------------+------------------+---------+--------+-------------+---------
----------------+-------------------------+-------------------------+--------------+------------------------------------------+---------------------------------------------------+--------------------------------------------
----------------------+----------------------------------------------------+--------------------------------------------------------------------------------+
| azuredevops:Build:1:1165300:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47543 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1165300 | app1 | SUCCESS | DONE   | PRODUCTION  | 2021-09-
22 18:29:59.000 | 2021-09-22 18:35:27.000 | 2021-09-22 18:36:13.000 |           46 | 8d5c383e05ca72b8ac24bdf605bd80e75e0b8714 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:915618:https://dev.azure.com/company/Company/_git/project  |
| azuredevops:Build:1:1168991:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"connection_id": 1, "scope_id": "f997d6c6-0ad1-4ad3-8787-53607f848ecb"} | _raw_azuredev
ops_builds |        37695 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1168991 | app2             | SUCCESS | DONE   | PRODUCTION  | 2021-09-
24 09:55:55.000 | 2021-09-24 09:59:43.000 | 2021-09-24 10:00:19.000 |           36 | 59a9a5ad8844ebacead9b27abde170f6911ac613 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1165300:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1169240:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47533 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1169240 | app3         | SUCCESS | DONE   | PRODUCTION  | 2021-09-
28 17:05:09.000 | 2021-09-28 17:08:09.000 | 2021-09-28 17:08:15.000 |            6 | 9b564ff18f94eeb710452494fa216749124d863e | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1177339:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1174834:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47541 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1174834 | app4             | SUCCESS | DONE   | PRODUCTION  | 2021-09-
27 13:53:11.000 | 2021-09-27 13:58:11.000 | 2021-09-27 13:58:13.000 |            2 | 8f991b408f1872533a676236c88abf76fe6e6d7a | refs/heads/feature/432300-switch-deploy-pipelines | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1168991:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1174837:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"connection_id": 1, "scope_id": "f997d6c6-0ad1-4ad3-8787-53607f848ecb"} | _raw_azuredev
ops_builds |        37693 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1174837 | app2             | SUCCESS | DONE   | PRODUCTION  | 2021-09-
27 13:54:41.000 | 2021-09-27 13:58:51.000 | 2021-09-27 13:59:16.000 |           25 | ba6c9e7e838278bab1f91e325b5d6555959a0bd2 | refs/heads/feature/432300-switch-deploy-pipelines | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1174834:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1174861:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"connection_id": 1, "scope_id": "f997d6c6-0ad1-4ad3-8787-53607f848ecb"} | _raw_azuredev
ops_builds |        37692 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1174861 | app2             | SUCCESS | DONE   | PRODUCTION  | 2021-09-
27 14:03:30.000 | 2021-09-27 14:07:30.000 | 2021-09-27 14:07:42.000 |           12 | 691bd67b729054af687926989f0f6f348234e6f3 | refs/heads/feature/432300-switch-deploy-pipelines | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1174837:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1176661:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47538 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1176661 | app5           | SUCCESS | DONE   | PRODUCTION  | 2021-09-
28 08:02:43.000 | 2021-09-28 08:17:45.000 | 2021-09-28 08:18:14.000 |           29 | da6efe6c78ada146844cf03916641e00fa03df96 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1174861:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1176749:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47537 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1176749 | app5           | SUCCESS | DONE   | PRODUCTION  | 2021-09-
28 08:52:55.000 | 2021-09-28 09:09:39.000 | 2021-09-28 09:09:47.000 |            8 | 41b97138261a871d6f7d98bbe7ff7f08a5f68ad3 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1176661:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1176876:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47536 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1176876 | app5           | SUCCESS | DONE   | PRODUCTION  | 2021-09-
28 09:47:24.000 | 2021-09-28 10:06:24.000 | 2021-09-28 10:06:53.000 |           29 | c234ffa3fcbcb67f4c22b3b4f4848542ca284983 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1176749:https://dev.azure.com/company/Company/_git/project |
| azuredevops:Build:1:1177076:https://dev.azure.com/company/Company/_git/project | 2023-06-16 08:43:14.238 | 2023-08-27 00:38:08.626 | {"ConnectionId":1,"ScopeId":"f997d6c6-0ad1-4ad3-8787-53607f848ecb"}      | _raw_azuredev
ops_builds |        47535 | prev_deployment_commit_id_enricher, | azuredevops:GitRepository:1:f997d6c6-0ad1-4ad3-8787-53607f848ecb | azuredevops:Build:1:1177076 | app5           | SUCCESS | DONE   | PRODUCTION  | 2021-09-
28 11:17:50.000 | 2021-09-28 11:35:30.000 | 2021-09-28 11:35:40.000 |           10 | 5c01ee147b1c2973820fd5893ab76feb93e148f1 | refs/heads/develop                                | azuredevops:GitRepository:1:f997d6c6-0ad1-4
ad3-8787-53607f848ecb | https://dev.azure.com/company/Company/_git/project | azuredevops:Build:1:1176876:https://dev.azure.com/company/Company/_git/project |
+--------------------------------------------------------------------------------+-------------------------+-------------------------+--------------------------------------------------------------------------+--------------
-----------+--------------+-------------------------------------+------------------------------------------------------------------+-----------------------------+------------------+---------+--------+-------------+---------
----------------+-------------------------+-------------------------+--------------+------------------------------------------+---------------------------------------------------+--------------------------------------------
----------------------+----------------------------------------------------+--------------------------------------------------------------------------------+
10 rows in set (0.00 sec)`

**Comment by klesh** [comment]
@KlemenDanfoss How about the `commits_diffs` table? It should contain `commits` between 2 consecutive `cicd_deployment_commits`  for the same `repo_url`, and your `pr.merge_commit_sha` should be included in the table, that is how we figured out which `cicd_deployment_commit` the PR belongs to.

**Comment by KlemenDanfoss** [comment]
@klesh 

`mysql> select count(*) from commits_diffs where new_commit_sha='8d5c383e05ca72b8ac24bdf605bd80e75e0b8714';
+----------+
| count(*) |
+----------+
|     2851 |
+----------+
1 row in set (0.00 sec)`

**Comment by klesh** [comment]
@KlemenDanfoss Sure, it would be nice. I'm in UTC+8, what is your timezone? you may contact me on the [slack](https://join.slack.com/t/devlake-io/shared_invite/zt-20envwfbk-JUTZ4z9jSeRnrvNhBFLg9w). My name is Klesh Wong

**Comment by KlemenDanfoss** [comment]
@klesh, I think this can be closed - #5996
